components/director/internal/domain/application/service.go

```diff
@@ -712,6 +714,60 @@ func (s *service) DeleteLabel(ctx context.Context, applicationID string, key str
 	return nil
 }
 
+// Merge Merges properties from Source Application into Destination Application, provided that the Destination's
+// Application does not have a value set for a given property. Then the Source Application is being deleted.
+func (s *service) Merge(ctx context.Context, destID, srcID string) (*model.Application, error) {
+	appTenant, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while loading tenant from context")
+	}
+
+	destApp, err := s.Get(ctx, destID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while getting destination application")
+	}
+
+	srcApp, err := s.Get(ctx, srcID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while getting source application")
+	}
+
+	srcBaseURL := str.PtrStrToStr(srcApp.BaseURL)
```

Comment:
You probably wanna make sure this works even if one of srcBaseURL or  destBaseURL has a trailing `/` symbol as part of the URL - so maybe trim it for both URLs just to be sure that both URLs are as equal as possible